File: 26.remove-duplicates-from-sorted-array.cpp
/*
 * @lc app=leetcode id=26 lang=cpp
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
#include <iostream>
#include <vector>

using namespace std;
class Solution {
 public:
  int removeDuplicates(vector<int>& nums) {
    int count = 0;

    if (nums.size() > 0) {
      for (int i = 1; i < nums.size(); i++) {
        if (nums[count] != nums[i]) {
          count++;
          nums[count] = nums[i];
        }
      }
      count++;
    }
    return count;
  }
};
// @lc code=end

File: Top Interview Questions/src/26.remove-duplicates-from-sorted-array.java
/*
 * @lc app=leetcode id=26 lang=java
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution {
  public int removeDuplicates(int[] nums) {
    int cursor = 0;

    for (int i = 1; i < nums.length; i++) {
      if (nums[i] != nums[cursor]) {
        nums[++cursor] = nums[i];
      }
    }
    
    return cursor + 1;
  }
}
// @lc code=end

